test/suites/greenboot.robot

```diff
@@ -0,0 +1,144 @@
+*** Settings ***
+Documentation       Tests related to greenboot
+
+Resource            ../resources/common.resource
+Resource            ../resources/systemd.resource
+Resource            ../resources/ostree-health.resource
+Resource            ../resources/ostree.resource
+Resource            ../resources/microshift-config.resource
+
+Suite Setup         Setup Suite
+Suite Teardown      Teardown Suite
+
+Test Tags           restart    slow
+
+
+*** Variables ***
+${LOCAL_WORKLOAD_FILE}          ../docs/config/busybox_running_check.sh
+${GREENBOOT_WORKLOAD_FILE}      /etc/greenboot/check/required.d/50_busybox_running_check.sh
+${GREENBOOT_CONFIG_FILE}        /etc/greenboot/greenboot.conf
+${GREENBOOT_CONFIG_CONTENT}     MICROSHIFT_WAIT_TIMEOUT_SEC=180
+${MANIFEST_SOURCE_DIR}          ./assets/kustomizations/greenboot/
+${MANIFEST_DIR}                 /etc/microshift/manifests
+${HOSTNAME_BIN_PATH}            ""
+
+
+*** Test Cases ***
+GreenbootHealth Run with User Workload
+    [Documentation]    Add user's workload and verify that the workload starts and greenboot is successful.
+    Greenboot Healthy
+    Add User Workload
+    Microshift Cleanup And Start
+
+    [Teardown]    Run Keywords
+    ...    Cleanup User Workload
+    ...    Microshift Cleanup And Start
+
+GreenbootHealth Microshift Service Failure
+    [Documentation]    Microshift Service failure
+    Greenboot Healthy
+    Microshift Disrupt Service
+    Cleanup MicroShift    "--keep-images"
+    Run Keyword And Expect Error    0 != 1
+    ...    Systemctl    enable    --now microshift
+    Run Keyword And Expect Error    0 != 1
+    ...    Greenboot Healthy
+
+    [Teardown]    Run Keywords
+    ...    Microshift Restore Service
+    ...    Microshift Cleanup And Start
+
+GreenbootHealth Microshift Pod Failure
+    [Documentation]    Microshift pod network failure
+    Greenboot Healthy
+    Microshift Disrupt Pod Network
+    Restart MicroShift
+    Run Keyword And Expect Error    0 != 1
+    ...    Greenboot Healthy
+
+    [Teardown]    Run Keywords
+    ...    Restore Default MicroShift Config
+    ...    Microshift Cleanup And Start
+
+
+*** Keywords ***
+Setup Suite
+    [Documentation]    Set up all of the tests in this suite
+    Check Required Env Variables
+    Login MicroShift Host
+    Setup Kubeconfig
+    Upload String To File    ${GREENBOOT_CONFIG_CONTENT}    ${GREENBOOT_CONFIG_FILE} # change the default timeout
+
+Add User Workload
+    [Documentation]    Upload User workload files to the MicroShift host
+    Put File    ${LOCAL_WORKLOAD_FILE}    /tmp/busybox_running_check.sh
+    ${stdout}    ${rc}=    Execute Command
+    ...    [ -f ${GREENBOOT_WORKLOAD_FILE} ] || sudo mv /tmp/busybox_running_check.sh ${GREENBOOT_WORKLOAD_FILE}
```

Comment:
Do we need `sudo` here? Does Robot's `sudo=True` no longer applies at this point?

Reply:
No .. Robo't only prepend `sudo` at the beginning of the expression.